micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/correlationid/CorrelationIdAspect.java

```diff
@@ -80,6 +90,17 @@ public Object call() throws Exception {
         });
     }
 
+    @Around("anyControllerOrRestControllerWithPublicWebAsyncTaskMethod()")
+    public Object wrapWebAsyncTaskWithCorrelationId(ProceedingJoinPoint pjp) throws Throwable {
+        final WebAsyncTask webAsyncTask = (WebAsyncTask) pjp.proceed();
+        log.debug("Wrapping webAsyncTask with correlation id [" + CorrelationIdHolder.get() + "]");
+
+        Field callableField = WebAsyncTask.class.getDeclaredField("callable");
```

Comment:
@nurkiewicz sure, I was thinking how to make it less brittle. Another option is to have a test for setting the callable field and break the build is the field name changes. May be even better than gracefully disabling the feature since the field name change is very unlikely in this case, IMO. WDYT?